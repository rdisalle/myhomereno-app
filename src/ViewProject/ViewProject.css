.ProjectItem {
    margin-left: 10px;
    margin-right: 10px;
}

@media only screen and (min-device-width: 768px) {
    .ProjectItem {
        width: 600px;
        margin-right: auto;
        margin-left: auto;
    }
    .Project_Link {
        margin-right: 20px;
    }
}

@media screen and (min-device-width: 1200px) {
    .ProjectItem {
        width: 1000px;
        margin-right: auto;
        margin-left: auto;
    }
    .Project_Link {
        margin-right: 20px;
    }
}