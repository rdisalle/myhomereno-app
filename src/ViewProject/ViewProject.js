import React, { Component } from 'react';
import './ViewProject.css';
import ProjectPage from '../ProjectPage/ProjectPage';

class ViewProject extends Component {
  render() {
    const project = this.props.info
        .map((info, key) => <ProjectPage {...info} key={key} />);
    return (
        <div>
        {project}
        </div>
    );
  }
}

ProjectPage.defaultProps = {
    info: []
  };
  
export default ViewProject;
