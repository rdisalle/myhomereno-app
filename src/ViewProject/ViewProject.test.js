import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewProject from './ViewProject';

describe(`ViewProjects component`, () => {
  const props = {
    projects: {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": 'My Project 1',
        "estimated_cost": '$10,000',
        "total_time": '4 weeks',
        "room":'Bathroom',
        "details": 'Entire bath reno',
        "type": 'Contactor',
        "status": 'In Progress',
        "summary": 'Getting bathroom redone',
      },
  };

  it('renders a .ViewProject by default', () => {
    const wrapper = shallow(<ViewProject />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a Projects with project prop', () => {
    const project = shallow(<ViewProject {...props} />)
      .find('project');
    expect(toJson(project)).toMatchSnapshot();
  });
});