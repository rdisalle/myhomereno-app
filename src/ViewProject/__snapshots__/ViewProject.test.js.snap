// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewProjects component renders a .ViewProject by default 1`] = `
<section
  className="ProjectItem"
>
  <ProjectPage
    delete={[Function]}
    details=""
    estimated_cost=""
    name=""
    room=""
    status=""
    summary=""
    total_time=""
    type=""
  />
</section>
`;

exports[`ViewProjects component renders a Projects with project prop 1`] = `null`;
